code/Common/src/main/java/net/selenate/common/comms/req/SeReqWindowFrameSwitch.java

```diff
@@ -1,61 +1,98 @@
 package net.selenate.common.comms.req;
 
-import java.util.List;
-import net.selenate.common.exceptions.SeNullArgumentException;
+import java.util.*;
+
 import net.selenate.common.SelenateUtils;
+import net.selenate.common.comms.*;
 
 public final class SeReqWindowFrameSwitch implements SeCommsReq {
   private static final long serialVersionUID = 45749879L;
 
-  private final List<Integer> framePath;
+  public static enum FrameSwitchMethod {
+    PATH,
+    NAME,
+    SELECTOR
+  }
+
+  private final FrameSwitchMethod method;
+  private final List<Integer>     framePath;
+  private final String            frameName;
+  private final SeElementSelector frameSelector;
 
   public SeReqWindowFrameSwitch(final List<Integer> framePath) {
-    this.framePath = framePath;
-    validate();
+    this.method        = FrameSwitchMethod.PATH;
+    this.framePath     = SelenateUtils.guardNull(framePath, "FramePath");
+    this.frameName     = null;
+    this.frameSelector = null;
+  }
+
+  public SeReqWindowFrameSwitch(final String frameName) {
```

Comment:
Only if you want to have 3 commands that do same thing (and then you dont need enum)
Or it could have one constructor but I totally discourage constructors with Object argument (since that is what List<integer>, String, and SeElementSelector have in common)

Client interface has 3 functions:
public void switchTo(final Integer... framePath) throws SeException;
public void switchTo(final String name) throws SeException;
public void switchTo(final IElementSelector selector) throws SeException;